lib/attributor/attribute.rb

```diff
@@ -228,78 +224,41 @@ def attributes
       type.attributes if @type_has_attributes ||= type.respond_to?(:attributes)
     end
 
-    # Validates stuff and checks dependencies
-    def validate(object, context = Attributor::DEFAULT_ROOT_CONTEXT)
-      raise "INVALID CONTEXT!! #{context}" unless context
-      # Validate any requirements, absolute or conditional, and return.
-
-      if object.nil? # == Attributor::UNSET
-        # With no value, we can only validate whether that is acceptable or not and return.
-        # Beyond that, no further validation should be done.
-        return validate_missing_value(context)
-      end
-
-      # TODO: support validation for other types of conditional dependencies based on values of other attributes
-
-      errors = validate_type(object, context)
-
-      # End validation if we don't even have the proper type to begin with
-      return errors if errors.any?
-
-      if options[:values] && !options[:values].include?(object)
-        errors << "Attribute #{Attributor.humanize_context(context)}: #{Attributor.errorize_value(object)} is not within the allowed values=#{options[:values].inspect} "
-      end
-
-      errors + type.validate(object, context, self)
+    def self.nullable_default
+      false
     end
 
-    def validate_missing_value(context)
-      raise "INVALID CONTEXT!!! (got: #{context.inspect})" unless context.is_a? Enumerable
-
-      # Missing attribute was required if :required option was set
-      return ["Attribute #{Attributor.humanize_context(context)} is required"] if options[:required]
-
-      # Missing attribute was not required if :required_if (and :required)
-      # option was NOT set
-      requirement = options[:required_if]
-      return [] unless requirement
-
-      case requirement
-      when ::String
-        key_path = requirement
-        predicate = nil
-      when ::Hash
-        # TODO: support multiple dependencies?
-        key_path = requirement.keys.first
-        predicate = requirement.values.first
+    # It is only nullable if there is an explicit null: true (or if it's not passed/set, and the default is true)
+    def self.nullable_attribute?(options)
+      if !options.key?(:null) || options[:null].nil?
```

Comment:
How can `options[:null].nil?` end up nil if we validate it below as being true or false?